pkg/command/command.go

```diff
@@ -240,81 +237,110 @@ func (c *Command) startCmd(cmd *exec.Cmd) (*execution, error) {
 		prOut, pwOut := io.Pipe()
 		trOut := io.TeeReader(stdout, pwOut)
 		srcOut = prOut
-
 		prErr, pwErr := io.Pipe()
 		trErr := io.TeeReader(stderr, pwErr)
 		srcErr = prErr
-
-		execution.wg.Add(2)
-
-		go func() {
-			defer execution.wg.Done()
-			defer pwOut.Close()
-			c.scanLog(trOut)
-		}()
-
-		go func() {
-			defer execution.wg.Done()
-			defer pwErr.Close()
-			c.scanLog(trErr)
-		}()
+		wg.Go(func() (err error) {
+			defer func() {
+				dErr := pwOut.Close()
+				if dErr != nil {
+					err = fmt.Errorf("can't close piper writer ")
+				}
+			}()
+			err = c.scanLog(trOut)
+			if err != nil {
+				err = fmt.Errorf("can't scan log: %w", err)
+			}
+			return
+		})
+		wg.Go(func() (err error) {
+			defer func() {
+				dErr := pwErr.Close()
+				if dErr != nil {
+					err = fmt.Errorf("can't close piper writer ")
+				}
+			}()
+			err = c.scanLog(trErr)
+			if err != nil {
+				err = fmt.Errorf("can't scan log: %w", err)
+			}
+			return nil
+		})
 	}
-
-	go func() {
-		if c.URLReportFileName != "" {
+	if c.StepName != "" {
+		cl := cumuluslog.NewCumulusLogger(c.StepName)
+		wg.Go(func() error {
 			var buf bytes.Buffer
 			br := bufio.NewWriter(&buf)
-			_, execution.errCopyStdout = piperutils.CopyData(io.MultiWriter(c.stdout, br), srcOut)
-			br.Flush()
-			handleURLs(buf.String(), c.URLReportFileName)
-		} else {
-			_, execution.errCopyStdout = piperutils.CopyData(c.stdout, srcOut)
-		}
-		execution.wg.Done()
-	}()
-
-	go func() {
-		if c.URLReportFileName != "" {
+			_, err := piperutils.CopyData(io.MultiWriter(c.stdout, br), srcOut)
```

Comment:
we have another routine below (line:313) which do the same with c.stdout. probably we don't need multiwriter here